### DISCUSSION TITLE
Libraries from other software

### DISCUSSION THREAD

**Discussion by NathanielakaJuan** [vdemichev/DiaNN#1104] [General]
1. Is it possible to use library from other software ? 
2. If yes, from which software library can be used ?
3. What are the extensions of file which can be used as libraries ? 
4. What should contain such file ? What kind of columns are required ?

I would like to use library from Spectronaut or MaxQuant.

**Comment by vdemichev** [comment]
Yes, possible: https://github.com/vdemichev/DiaNN?tab=readme-ov-file#spectral-library-formats

**Reply by NathanielakaJuan** [reply]
Yeah, that part I read but it seems like using msms.txt from MQ does not work and also evidence.txt file. I was just curious if someone has tried that also and can tell me what is wrong with these outputs. 

I am just curious what exactly columns are needed to be a "proper" library for DIA-NN.

**Reply by vdemichev** [reply]
Yes, you can expand 'in detail' by the link above, it lists the columns.
Strange that does not work with msms.txt, what do you see in the output when you try it?

**Reply by NathanielakaJuan** [reply]
_WARNING: support for MaxQuant search results is experimental; assuming all cysteines are carbamidomethylated; MaxQuant does not include fixed modifications in modified peptide names, so all other fixed modifications will need to be manually annotated
Library contains q-values but there are no decoys included; if the library was generated with an old version of DIA-NN, this might lead to less reliable run-specific protein FDR when using it; ideally, the library needs to be generated using DIA-NN 1.9 or later_

That's a warning which I get and then DIA-NN exited. It just stops after 30 secs.

**Reply by vdemichev** [reply]
Warnings are fine.

> That's a warning which I get and then DIA-NN exited.

How does the log look like?

**Reply by NathanielakaJuan** [reply]
U mean this: 

__diann.exe --f "D:\DIA-NN Calculations\XXXX\M6_250_DIA_1_1_1_691.d
" --f "D:\DIA-NN Calculations\XXXX\M8_250_DIA_1_1_1_692.d
" --f "D:\DIA-NN Calculations\XXXX\M9_250_DIA_1_1_1_694.d
" --f "D:\DIA-NN Calculations\XXXX\M10_250_DIA_1_1_1_695.d
" --lib "D:\XXXX\txt\msms.txt" --threads 64 --verbose 1 --out "D:\DIA-NN Calculations\XXXX\Results\report.tsv" --qvalue 0.01 --matrices  --out-lib "D:\DIA-NN Calculations\Library\XXXX4xMQ.parquet" --gen-spec-lib --unimod4 --mass-acc 10 --mass-acc-ms1 10 --reanalyse --relaxed-prot-inf --rt-profiling 
DIA-NN 1.9.1 (Data-Independent Acquisition by Neural Networks)
Compiled on Jul 15 2024 15:40:36
Current date and time: Mon Jul 22 23:46:43 2024
CPU: AuthenticAMD AMD Ryzen Threadripper 7980X 64-Cores
SIMD instructions: AVX AVX2 AVX512CD AVX512F FMA SSE4.1 SSE4.2 SSE4a 
Logical CPU cores: 64
Thread number set to 64
Output will be filtered at 0.01 FDR
Precursor/protein x samples expression level matrices will be saved along with the main report
A spectral library will be generated
Cysteine carbamidomethylation enabled as a fixed modification
A spectral library will be created from the DIA runs and used to reanalyse them; .quant files will only be saved to disk during the first step
Heuristic protein grouping will be used, to reduce the number of protein groups obtained; this mode is recommended for benchmarking protein ID numbers, GO/pathway and system-scale analyses
The spectral library (if generated) will retain the original spectra but will include empirically-aligned RTs
Mass accuracy will be fixed to 1e-05 (MS2) and 1e-05 (MS1)
WARNING: protein inference is enabled but no FASTA provided - is this intended?

4 files will be processed
[0:00] Loading spectral library D:\XXXX\txt\msms.txt
WARNING: support for MaxQuant search results is experimental; assuming all cysteines are carbamidomethylated; MaxQuant does not include fixed modifications in modified peptide names, so all other fixed modifications will need to be manually annotated
Library contains q-values but there are no decoys included; if the library was generated with an old version of DIA-NN, this might lead to less reliable run-specific protein FDR when using it; ideally, the library needs to be generated using DIA-NN 1.9 or later

DIA-NN exited
DIA-NN-plotter.exe "XXXX"
PDF report will be generated in the background__


Sorry for stupid questions. This is a first day of trying DIA-NN.

**Comment by vdemichev** [comment]
Can you please share msms.txt? I will find out what is the cause and fix in 1.9.2.

About library creation in general, we recommend:
- FragPipe, when creating a library from DDA data
- DIA-NN itself, from DIA data

**Reply by NathanielakaJuan** [reply]
I will upload it somewhere and I will choose the smallest one. I have tried like 5 different msms.txt. However, each of them is 1 gb+. 

I run the same parameters, the same DIA files with library made from fasta using DIA-NN and it worked fine. I just replaced the library with msms.txt and run once again. That's how it crushes. I guess fasta or anything else is not required ?

**Reply by vdemichev** [reply]
It used to support msms.txt. Maybe support got broken in 1.9 or maybe MaxQuant team changed the format. Either way, this will be easy to fix.

**Reply by vdemichev** [reply]
7z should make it a lot smaller for an upload

**Reply by NathanielakaJuan** [reply]
https://we.tl/t-p8qaYYrbeI

uploaded

**Reply by vdemichev** [reply]
Thank you, found the error, will be fixed in 1.9.2.

**Reply by NathanielakaJuan** [reply]
Super, thanks!
